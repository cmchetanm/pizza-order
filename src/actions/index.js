export function selectDough(dough) {
  // selectDough is an ActionCreator, it needs to return an action,
  // an object with a type property.
  return {
    type: "DOUGH_SELECTED",
    payload: dough
  };
}

export function selectIngredient(ingredient) {
  // selectIngredient is an ActionCreator, it needs to return an action,
  // an object with a type property.
  return {
    type: "INGREDIENT_SELECTED",
    payload: ingredient
  };
}

export function removeIngredient(ingredient) {
  // selectIngredient is an ActionCreator, it needs to return an action,
  // an object with a type property.
  return {
    type: "INGREDIENT_REMOVED",
    payload: ingredient
  };
}
