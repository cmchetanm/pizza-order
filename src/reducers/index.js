import { combineReducers } from "redux";
import DoughTypes from "./reducer_dough_types";
import activeDoughType from './reducer_active_dough';
import Ingredients from './reducer_ingredient_list';
import ActiveIngredients from './reducer_active_ingredient'

const rootReducer = combineReducers({
  doughTypes: DoughTypes,
  activeDoughTypes: activeDoughType,
  ingredients: Ingredients,
  activeIngredients: ActiveIngredients
});

export default rootReducer;
