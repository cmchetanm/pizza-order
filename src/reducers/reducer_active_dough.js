const initialAction = {
  type: null
}

export default function(state = null, action = initialAction) {
  switch (action.type) {
    case "DOUGH_SELECTED":
      return action.payload;
    default:
      return state
  }
}
