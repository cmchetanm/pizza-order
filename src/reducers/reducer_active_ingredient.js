// State argument is not application state, only the state
// this reducer is responsible for
const initialState = {
  totalPrice: null,
  ingredients: []
}

const initialAction = {
  type: null
}

function removeByKey(array, params){
  array.some(function(item, index) {
    return (array[index].id === params.id) ? !!(array.splice(index, 1)) : false;
  });
  return array;
}

function addIngredient(state= null, action) {
  state.ingredients.push(action.payload)
  state.price = state.ingredients.reduce((prev, next) => prev + next.price, 0);
  return {totalPrice: state.price, ingredients: state.ingredients};
}

function removeIngredient(state= null, action) {
  state.ingredients = removeByKey(state.ingredients, action.payload)
  state.price = state.ingredients.reduce((prev, next) => prev + next.price, 0);
  return {totalPrice: state.price, ingredients: state.ingredients};
}

export default function(state = initialState, action = initialAction) {
  switch (action.type) {
    case "INGREDIENT_SELECTED":
      return addIngredient(state, action);
    case 'INGREDIENT_REMOVED':
      return removeIngredient(state, action);
    default:
      return state
  }
}
