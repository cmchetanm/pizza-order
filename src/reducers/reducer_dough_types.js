export default function() {
  return [
    {
      id: 1,
      name: 'Thin',
      description: 'Thin and crusty pizza dough',
      price: 0.5

    },
    {
      id: 2,
      name: 'Thick',
      description: 'Thick and crusty pizza dough',
      price: 1
    }
  ];
}
