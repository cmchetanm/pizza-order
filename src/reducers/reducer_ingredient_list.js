export default function() {
  return [
    {
      id: 1,
      name: 'pepperoni',
      displayName: 'Pepperoni',
      price: 1
    },
    {
      id: 2,
      name: 'anchovies',
      displayName: 'Anchovies',
      price: 10
    },
    {
      id: 3,
      name: 'lobster',
      displayName: 'Lobstah',
      price: 25
    },
    {
      id: 4,
      name: 'truffle oil',
      displayName: 'truffle oil',
      price: 100
    }
  ];
}
