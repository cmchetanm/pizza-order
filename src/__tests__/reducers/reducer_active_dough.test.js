import reducers from "../../reducers";
import { selectDough } from "../../actions/index";

const doughType =  {
  activeDoughTypes:  {
    id: 1,
    name: 'Thin',
    description: 'Thin and crusty pizza dough',
    price: 0.5
  }
}

const currentDoughType = {
  activeDoughTypes:  {
    id: 1,
    name: 'Thin',
    description: 'Thin and crusty pizza dough',
    price: 0.5
  }
}

describe('doughType reducer', () => {
  test('counter should return the initial state', () => {
    expect(reducers(undefined, {})).toMatchSnapshot();
  });

  test('doughType selectDough', () => {
    expect(reducers(currentDoughType, selectDough(doughType))).toMatchSnapshot();
  });
});
