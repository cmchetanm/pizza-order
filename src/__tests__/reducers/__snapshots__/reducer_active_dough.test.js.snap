// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doughType reducer counter should return the initial state 1`] = `
Object {
  "activeDoughTypes": null,
  "activeIngredients": Object {
    "ingredients": Array [],
    "totalPrice": null,
  },
  "doughTypes": Array [
    Object {
      "description": "Thin and crusty pizza dough",
      "id": 1,
      "name": "Thin",
      "price": 0.5,
    },
    Object {
      "description": "Thick and crusty pizza dough",
      "id": 2,
      "name": "Thick",
      "price": 1,
    },
  ],
  "ingredients": Array [
    Object {
      "displayName": "Pepperoni",
      "id": 1,
      "name": "pepperoni",
      "price": 1,
    },
    Object {
      "displayName": "Anchovies",
      "id": 2,
      "name": "anchovies",
      "price": 10,
    },
    Object {
      "displayName": "Lobstah",
      "id": 3,
      "name": "lobster",
      "price": 25,
    },
    Object {
      "displayName": "truffle oil",
      "id": 4,
      "name": "truffle oil",
      "price": 100,
    },
  ],
}
`;

exports[`doughType reducer doughType selectDough 1`] = `
Object {
  "activeDoughTypes": Object {
    "activeDoughTypes": Object {
      "description": "Thin and crusty pizza dough",
      "id": 1,
      "name": "Thin",
      "price": 0.5,
    },
  },
  "activeIngredients": Object {
    "ingredients": Array [],
    "totalPrice": null,
  },
  "doughTypes": Array [
    Object {
      "description": "Thin and crusty pizza dough",
      "id": 1,
      "name": "Thin",
      "price": 0.5,
    },
    Object {
      "description": "Thick and crusty pizza dough",
      "id": 2,
      "name": "Thick",
      "price": 1,
    },
  ],
  "ingredients": Array [
    Object {
      "displayName": "Pepperoni",
      "id": 1,
      "name": "pepperoni",
      "price": 1,
    },
    Object {
      "displayName": "Anchovies",
      "id": 2,
      "name": "anchovies",
      "price": 10,
    },
    Object {
      "displayName": "Lobstah",
      "id": 3,
      "name": "lobster",
      "price": 25,
    },
    Object {
      "displayName": "truffle oil",
      "id": 4,
      "name": "truffle oil",
      "price": 100,
    },
  ],
}
`;
